tests/test_jointcalRunner.py

```diff
@@ -0,0 +1,141 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import inspect
+import os.path
+import unittest
+import unittest.mock
+
+import lsst.log
+import lsst.utils
+
+import lsst.daf.persistence
+import lsst.jointcal
+from lsst.meas.algorithms import DatasetConfig
+import lsst.pipe.base
+
+
+# for MemoryTestCase
+def setup_module(module):
+    lsst.utils.tests.init()
+
+
+class TestJointcalRunner(lsst.utils.tests.TestCase):
+    """Test that JointcalRunner calls JointcalTask with appropriate arguments."""
+    @classmethod
+    def setUpClass(cls):
+        cls.data_dir = os.path.join(lsst.utils.getPackageDir('jointcal'), 'tests/data')
+
+    def setUp(self):
+        def mock_get(arg):
+            if arg == 'verify_job_filename':
+                return 'some/path/filename.job'
+            elif arg == 'ref_cat_config':
+                result = unittest.mock.Mock(DatasetConfig)
+                result.indexer = DatasetConfig().indexer
+                return result
+            else:
+                return None
+
+        def mock_dataRef(datasetType, level=None, dataId=None):
+            ref = lsst.daf.persistence.ButlerDataRef(None, dataId)
+            ref.datasetExists = lambda: True
+            return ref
+
+        def mock_subset(datasetType, dataId=None):
+            return [mock_dataRef(datasetType, dataId=dataId)]
+
+        butlerPatcher = unittest.mock.patch("lsst.daf.persistence.Butler", autospec=True)
+        self.butler = butlerPatcher.start()
+        self.addCleanup(butlerPatcher.stop)
+        self.butler.return_value.get = mock_get
+        self.butler.return_value.subset = mock_subset
+        self.butler.return_value.dataRef = mock_dataRef
+        self.butler.return_value.getKeys.return_value = {'visit': int, 'ccd': int}
+        # these two are static methods
+        self.butler.getMapperClass.return_value.getPackageName.return_value = 'jointcal'
+        self.butler.getMapperClass.return_value.getCameraName.return_value = 'cfht'
+
+    def prep_jointcal(self, tracts=None):
+        """Prepare a jointcal mock to be called by JointcalRunner.
+
+        We use the data/cfht_minimal to provide a "real" refcat, and use the
+        ids associated with it, even though no real butler is involved.
+
+        Parameters
+        ----------
+        tracts : `list` [`int`]
+            List of tracts to build DataRefs for.
+        """
+        configfile = os.path.join(lsst.utils.getPackageDir('jointcal'), 'tests/config/config.py')
+
+        input_dir = os.path.join(self.data_dir, 'cfht_minimal')
+        # the calling method is one step back on the stack: use it to specify the output repo.
+        caller = inspect.stack()[1].function
+        output_dir = os.path.join('.test', self.__class__.__name__, caller)
```

Comment:
Are you doing these shenanigans to aid debugging the test?   Why `.test/`? Why not a temp directory? Is a directory actually created or is that mocked too? Can you be more explicit in your description that you want to name this directory after the test method that is being executed rather than saying "the calling method"?

Reply:
Good point: I copied this from the other jointcal tests where not having them auto-deleted (i.e. `tempfile`) has been very useful. In this case, `tempfile` is definitely appropriate.